lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -1619,6 +1619,16 @@ private static JsonNode validateDefault(String fieldName, Schema schema, JsonNod
     return defaultValue;
   }
 
+  /**
+   * Checks if a JSON value matches the schema.
+   *
+   * @param jsonValue a value to check against the schema
+   * @return true if the value is valid according to this schema
+   */
+  public boolean isValidValue(JsonNode jsonValue) {
+    return isValidDefault(this, jsonValue);
+  }
```

Comment:
:+1: use standard method instead of static (_it may replace private static boolean isValidDefault in near future_), and avoid this long switch/case on schema/type.
About default value on union type, I wonder if it couldn't be even better to change logic and get the first compatible Schema of union with the value (And throw exception if none) ?
I did this code in local (_and partially copy yours_), and it seems to work fine (at least for unit test). I will put more details in linked JIRA Ticket in few minutes.

Reply:
thx for the feedback, can u give me some points what `standard method` you are talking about?

Reply:
Just was able to take a look at your PR. It looks like a even better solution, because it "fixes" the problem on a lower level.

Would you say that this PR is now obsolete?

Reply:
Is there any possible way we can rewrite this without adding or exposing the `isValidValue(JsonNode ...)` method?

There was a reason the isValidDefault method is private: we **really** don't want to be exposing the jackson internals in public methods and (if I remember correctly).  There was a fair amount of work deleting these methods in the past, in order to _one day_ replace the jackson internals.

If anyone has a better memory than me, please feel to chime in!